src/test/java/com/ordernow/backend/auth/controller/v2/AuthControllerIntegrationTest.java

```diff
@@ -0,0 +1,142 @@
+package com.ordernow.backend.auth.controller.v2;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ordernow.backend.auth.model.dto.LoginRequest;
+import com.ordernow.backend.auth.model.entity.Customer;
+import com.ordernow.backend.auth.model.entity.Role;
+import com.ordernow.backend.auth.repository.UserRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import lombok.extern.slf4j.Slf4j;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles("test")
+@Slf4j
+public class AuthControllerIntegrationTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private final BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder(12);
+
+    private final String TEST_REGISTERED_MAIL = "registered@example.com";
+    private final String TEST_REGISTERED_PASSWORD = "registeredpassword";
```

Comment:
don't use global variable here, because every testing should independent.
set in testing function.